### PR TITLE
(MODULES-3655) Add hash2plist function

### PR DESCRIPTION
**Pull request by keeleysam**

### CODE REVIEW

**Comment by keeleysam** [comment]
The tests pass on the latest version of Puppet, but this will not work on Puppet prior to version 4.4.0 as `puppet/util/plist` wasn't yet present.   I'm not sure how to confine this properly to deal with that limitation.

File: lib/puppet/parser/functions/hash2plist.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# 
+# hash2plist.rb
+#
+
+require 'puppet/util/plist'
+
+module Puppet::Parser::Functions
+  newfunction(:hash2plist, :type => :rvalue , :doc => <<-EOS
+Takes a hash as an input and returns a plist as a string useful
```
Comment (igalic) [inline lib/puppet/parser/functions/hash2plist.rb]: `as a string useful` ⇒ `as a string. Useful`

however: an api documentation shouldn't have to justify a function's existence.
it should do so, by explaining what it does.

**Comment by igalic** [comment]
this function seems oddly (platform?) specific for a "standard" library…

**Comment by wilson208** [comment]
@keeleysam I am closing this PR because it hasnt seen any activity in a number of months. If you want to work on this again please feel free to re-open. Thanks!

**Pull request closed without merging**